cpp/src/arrow/compute/kernels/vector_replace.cc

```diff
@@ -154,9 +172,10 @@ Status ReplaceWithArrayMask(KernelContext* ctx, const ArrayData& array,
   uint8_t* out_values = output->buffers[1]->mutable_data();
   const uint8_t* mask_bitmap = mask.MayHaveNulls() ? mask.buffers[0]->data() : nullptr;
   const uint8_t* mask_values = mask.buffers[1]->data();
-  const bool replacements_bitmap = replacements.is_array()
-                                       ? replacements.array()->MayHaveNulls()
-                                       : !replacements.scalar()->is_valid;
+  const bool replacements_bitmap =
```

Comment:
This is overloading the meaning of this flag - making it misleading.

Reply:
solved